### PR TITLE
feat(redux): exclude update product event for next

### PR DESCRIPTION
**Pull request by danielbento92**
## Description

- Adjust middlewares:
  - Changing size on cart or wishlist pages will trigger only change size
   - Changing quantity always add/remove products
   events and change quantity event
    - When add product that was already on bag, should not trigger
    change quantity event.
- Add tests

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by ruifcnunes — APPROVED** [review]

File: packages/redux/src/analytics/middlewares/__tests__/bag.test.ts [hunk 1]
```diff
@@ -393,80 +396,119 @@ describe('analyticsBagMiddleware', () => {
...
-        oldSizeScaleId: sizes[0]?.scale,
-        price: priceWithDiscount,
-        priceWithoutDiscount,
-        quantity: 3,
-        size: sizes[1]?.name,
-        sizeId: sizes[1]?.id,
-        sizeScaleId: sizes[1]?.scale,
-        sku,
-        variant: colorName,
+      // expect analytics PRODUCT_ADDED_TO_CART event to be triggered
```
Comment (boliveira) [inline packages/redux/src/analytics/middlewares/__tests__/bag.test.ts:428]: This comment does not match the code

File: packages/redux/src/analytics/middlewares/__tests__/bag.test.ts [hunk 2]
```diff
@@ -393,80 +396,119 @@ describe('analyticsBagMiddleware', () => {
...
-      ]);
+      const bagItem = bagMockData.mockBagItem().getItem({ size });
 
-      await simplifiedStore.dispatch({
+      await store.dispatch({
         type: bagActionTypes.UPDATE_BAG_ITEM_SUCCESS,
         payload: {
           result: { id: bagMockData.mockBagId },
           entities: {
+            size: {
```
Comment (boliveira) [inline packages/redux/src/analytics/middlewares/__tests__/bag.test.ts:413]: Does this entity exist? Looking at the bag item schema this does not seem to be created

File: packages/redux/src/analytics/middlewares/wishlist.ts [hunk 3]
```diff
@@ -283,27 +283,47 @@ export function analyticsWishlistMiddleware(
...
-          EventTypes.PRODUCT_UPDATED_WISHLIST,
-          analyticsData,
-        );
+        analyticsInstance.track(EventTypes.PRODUCT_UPDATED_WISHLIST, {
+          ...data,
+        });
+
+        let eventType = null;
+
+        // Check if the quantity difference is less than it was in bag - use PRODUCT_REMOVED_FROM_CART in that case
```
Comment (boliveira) [inline packages/redux/src/analytics/middlewares/wishlist.ts:309]:
```suggestion
        // Check if the quantity difference is less than it was in bag - use PRODUCT_REMOVED_FROM_WISHLIST in that case
```

File: packages/redux/src/analytics/middlewares/__tests__/wishlist.test.ts [hunk 4]
```diff
@@ -768,6 +832,133 @@ describe('analyticsWishlistMiddleware', () => {
...
+      const size = { id: 999, name: 'XXL' };
+      const mockStore = mockSimplifiedStore(getMockState(), [
+        analyticsWishlistMiddleware(analytics),
+      ]);
+
+      await mockStore.dispatch({
+        type: actionTypes.UPDATE_WISHLIST_ITEM_SUCCESS,
+        payload: {
+          entities: {
+            size: {
```
Comment (boliveira) [inline packages/redux/src/analytics/middlewares/__tests__/wishlist.test.ts:846]: This entity does not seem to exist in the payload of this action.

**Review by nelsonleite — APPROVED** [review]

**Review by boliveira — APPROVED** [review]

**Pull request merged**